lib/storage/src/content_manager/consensus/persistent.rs

```diff
@@ -275,6 +310,34 @@ mod serialize_peer_addresses {
     }
 }
 
+mod serialize_peer_metadata {
+    use std::collections::HashMap;
+    use std::sync::Arc;
+
+    use parking_lot::RwLock;
+    use serde::{self, Deserialize, Deserializer, Serialize, Serializer};
+
+    use crate::types::{PeerMetadata, PeerMetadataById};
+
+    pub fn serialize<S>(
+        metadata: &Arc<RwLock<PeerMetadataById>>,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        metadata.read().serialize(serializer)
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Arc<RwLock<PeerMetadataById>>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let metadata: HashMap<u64, PeerMetadata> = PeerMetadataById::deserialize(deserializer)?;
+        Ok(Arc::new(RwLock::new(metadata)))
+    }
+}
```

Comment:
There are `serde/rc` and `parking_lot/serde` features. Or these don't work somehow? 🤔

(Didn't check `serialize_peer_addresses`, but maybe we can get rid of it as well, if it's similar to `serialize_peer_metadata`.)

Reply:
> There are serde/rc and parking_lot/serde features. Or these don't work somehow? 🤔

I copied the logic for URLs. I'll check to see if I can reduce it with predefined serde implementations.

Reply:
Adding the `serde` feature to `parking_lot` doesn't work sadly.

<details>
<summary>Cargo errors</summary>

```rust
 1  error[E0277]: the trait bound `std::sync::Arc<parking_lot::lock_api::RwLock<parking_lot::RawRwLock, s
 td::collections::HashMap<u64, types::PeerMetadata>>>: types::_::_serde::Serialize` is not satisfied
     --> lib/storage/src/content_manager/consensus/persistent.rs:27:17
     --> /home/timvisee/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.197/src/ser/mod.rs
 :1865:12
 2  error[E0277]: the trait bound `std::sync::Arc<parking_lot::lock_api::RwLock<parking_lot::RawRwLock, s
 td::collections::HashMap<u64, types::PeerMetadata>>>: types::_::_serde::Deserialize<'_>` is not satisfie
 d
     --> lib/storage/src/content_manager/consensus/persistent.rs:43:30
     --> /home/timvisee/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.197/src/de/mod.rs:
 1726:12
 3  error[E0277]: the trait bound `std::sync::Arc<parking_lot::lock_api::RwLock<parking_lot::RawRwLock, s
 td::collections::HashMap<u64, types::PeerMetadata>>>: types::_::_serde::Deserialize<'_>` is not satisfie
 d
     --> lib/storage/src/content_manager/consensus/persistent.rs:43:30
     --> /home/timvisee/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.197/src/de/mod.rs:
 1865:12
```

</details>

Reply:
You may have to also enable `rc` feature of `serde` crate?

Sorry if you've already tried it, I'm just totally *not* reading this logs. :D

Both `Arc` *and* `parking_lot` implement serde under a feature, so you have to enable *both* of them.

Reply:
Missed `rc`. Works now. Thanks!

Reply:
We cannot do it on the old one though. `Uri` does not implement serde traits. 

There is [`http_serde`](https://docs.rs/http-serde/latest/http_serde/), but we cannot use them because we've wrapped the types. So I'll leave the existing module in place.